src/client/cl_screen.c

```diff
@@ -387,6 +387,10 @@ void Cl_UpdateScreen(void) {
 		}
 	}
 
+	if (cls.state == CL_ACTIVE) {
+		R_AddStains();
+	}
+
```

Comment:
It can be moved up, but no it can't go in R_DrawView because this is not part of the view, this updates the stain framebuffers which are 2D (so Setup2D needs to be called first).